notifications_utils/field.py

```diff
@@ -130,19 +131,22 @@ def format_match(self, match):
 
     def replace_match(self, match):
```

Comment:
I think it's going to be very valuable to add some decent commenting about what's happening to the values and content as we're going through here.

Specifically for this one, that if we're replacing a match, then the content will have been sanitized, including placeholder names. Which is why we need to sanitize the placeholder names in the values dict.